view/propertyEditor.js

```diff
@@ -245,77 +244,141 @@ ACS.propertyEditor = function (modelList, // ACS.modelList
 				cell.appendChild(dropdownList);
 			}
 
-			//generate checkbox field for boolean
-			if (comboValues === '' && typetemp === ACS.dataType.BOOLEAN) {
+			if (isDynProp) {
+
+				log.debug('downloading (dynamic) properties list');
 				cell = row[h + 1].insertCell(1);
-				boolInput = null;
-				boolInput = document.createElement("INPUT");
-				boolInput.setAttribute("type", "checkbox");
-				boolInput.setAttribute("value", valtemp);
-				if ((valtemp === "true") || (valtemp === "True")) {
-					boolInput.setAttribute("checked", true);
+
+				if (ACS.areStatus.getSynchronised() === true) {
+
+					var id = propertyEdPanelCaptions.innerHTML;
+					var key = propName;
+					var currentValue = valtemp || "";
+					var elementId = h + "/1/" + valtemp;
+	
+					fetchDynProperties(cell, elementId, currentValue, id, key);
+					function fetchDynProperties(cellToAdd, elementId, currentValue, id, key) {
+						getRuntimeComponentPropertyList(function(data, httpStatus) {
+							var entries = parseEntries(data);
+							if (entries.length === 0) {
+								insertInputElement(cellToAdd, currentValue, elementId);
+							} else {
+								entries.unshift("");
+								insertSelectionElements(cellToAdd, entries, currentValue, elementId);
+							}
+						}, function(HTTPstatus, AREerrorMessage) {
+							console.error('check if ARE is running and connected with WebACS');
+							console.error('error: ' + AREerrorMessage);
+							insertInputElement(cellToAdd, currentValue, elementId);
+						}, id, key);
+					};
+
+					function insertInputElement(cell, currentValue, elementId) {
+						console.warn('no dynamic properties from ARE - using textfield instead of combobox');
+						var inputElement = document.createElement('input');
+						inputElement.value = currentValue;
+						inputElement.setAttribute("id", elementId);
```

Comment:
Great - looks quite clean now 👍 
one thing I just noticed: in case of the plain input element there is no `addEventListener()` call like in the case with the combobox. I don't know if it is needed?! Beside of this I think we finally can merge 😃

Reply:
Should probably be in there. Maybe I've missed it or I've rationalized during the changes of this pr. :smile: Anyway, I'll add it. What is strange is that during testing, I've changed the value of this input element (with disconnected ARE), stored the model and loaded it afterwards. And it worked.

I've apparently also overlooked the outermost else branch at lines 305-308. Should call `insertInputElement()` there too. Shouldn't I?

Reply:
> I've changed the value of this input element (with disconnected ARE), stored the model and loaded it afterwards. And it worked.

I think the `change` event listener is for live updates of the values, if connected with ARE and the model is running on ARE. For storing/loading the model it should not make a difference.

> ... at lines 305-308. Should call insertInputElement() there too. Shouldn't I?

yes, makes sense ;)

Reply:
It does not seem to work any more.
I did the following steps:
1. create a model with Knx plugin
2. upload model
3. start model
4. select plugin

There is no combobox and there are error messages in the console:
![image](https://user-images.githubusercontent.com/4621810/55815374-8ca84680-5af0-11e9-959a-dac75900fe78.png)

Reply:
I just tried the 4 steps you mention. I cannot reproduce the described error. My model is, however, the most simple case - just the KNX plugin :smile:. Have you used a more sophisticated model?

![image](https://user-images.githubusercontent.com/13609768/55828863-a4da8e80-5b0d-11e9-9645-b7903470d478.png)

What I did was following:
* place KNX plugin
* upload model
* start model
* select combobox (dyn. prop.)

Can you show me your model tomorrow? Or maybe add it to your comment - but I'm not sure, if you can add attachments to post after all.

Reply:
ok, sorry my mistake!!
I forgot to start the corresponding ARE with the new REST call, see https://github.com/asterics/AsTeRICS/pull/308

Works like a charme :+1: 
Let's merge!